packages/react-refresh/src/__tests__/__snapshots__/ReactFreshBabelPlugin-test.js.snap

```diff
@@ -108,21 +122,21 @@ exports[`ReactFreshBabelPlugin generates signatures for function expressions cal
 var _s = $RefreshSig$(),
     _s2 = $RefreshSig$();
 
-export const A = React.memo(_c2 = React.forwardRef(_c = _s((props, ref) => {
+export const A = _s(React.memo(_c2 = _s(React.forwardRef(_c = _s((props, ref) => {
   _s();
 
   const [foo, setFoo] = useState(0);
   React.useEffect(() => {});
   return <h1 ref={ref}>{foo}</h1>;
-}, "useState{[foo, setFoo](0)}\\nuseEffect{}")));
+}, "useState{[foo, setFoo](0)}\\nuseEffect{}")), "useState{[foo, setFoo](0)}\\nuseEffect{}")), "useState{[foo, setFoo](0)}\\nuseEffect{}");
 _c3 = A;
-export const B = React.memo(_c5 = React.forwardRef(_c4 = _s2(function (props, ref) {
+export const B = _s2(React.memo(_c5 = _s2(React.forwardRef(_c4 = _s2(function (props, ref) {
```

Comment:
I'm porting this PR to the TypeScript version.
I aggressively detect if the function needs to be hashed appears at the callee position of the CallExpression (which means only literal IIFE (`(() => {})(...args)`)) instead of allowing it appears in the argument position (`memo(f)`). Is that OK?

Reply:
Not sure I understand the question.

Reply:
So in this PR, it seems there are some irrelated (or false positive) changes in the snapshot:

https://github.com/facebook/react/pull/21104/files/00178dbaf7b5af453807a9d57724568b7c67759b?file-filters%5B%5D=.snap#diff-946346b4b3bb1833d001b16c63e3b733545cb5bbc651c2bfb25c37f664fd1a6dL40-R46

And in [the TS version](https://github.com/Jack-Works/react-refresh-transformer/pull/5/files) there are no such snapshot changes, only the new test case


```js
while (item) {
    ;((item) => {
        useFoo()
    })(item)
}
```

I'd like to know if those changes are _really_ false positive or I should port those behaviors to make it work correctly?

Reply:
I'll release it for now

Reply:
Why do you call these changes false positive? Can you describe what you expected vs what you saw?

Reply:
So as the PR title describes, it add support for IIFEs. But these baseline changes are not IIFEs, just a normal function call

Reply:
I think there's a misunderstanding.

The change in this PR has nothing to do with IIFEs. It only relates to higher-order components that _call_ rather than _render_ the passed function.

The IIFE test case is not something that's related to Hooks or Fast Refresh. It's just that we had a case like this in the codebase that was accidentally getting detected as a Hook call (because of the `use` prefix). My change broke the semantics of that code. So I added a regression test that verifies that code like this does not get broken.

Reply:
So what's the actual test case? Can you give a minimal example so I can fix that?

Reply:
The test showing _incorrect_ output is https://github.com/facebook/react/pull/21104/commits/d2a2045ba741ee33906ad0b4c6fd5ce6b8563fed. Then https://github.com/facebook/react/pull/21104/commits/070262b092c0925d255f907c5d187e35b4ba67b5 shows the fix and the correct output.

Reply:
Concretely, the problem was that walking upwards was only correct as long as the inner node represents the argument. If the child node was the _callee itself_ we should stop walking upwards.